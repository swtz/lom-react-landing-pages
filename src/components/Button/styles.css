.btn {
  width: 100%;
  background: darkblue;
  color: white;
  border: none;
  padding: 15px 30px;
  cursor: pointer;
}

.btn:disabled {
  background: #888;
  cursor: not-allowed;
}
