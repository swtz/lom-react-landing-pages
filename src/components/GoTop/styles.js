import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    position: fixed;
    background: ${theme.colors.primaryColor};
    color: ${theme.colors.white};
    display: flex;
    align-items: center;
    justify-content: center;
    bottom: 2rem;
    right: 2rem;
    width: 4rem;
    height: 4rem;
    z-index: 6;
  `}
`;
