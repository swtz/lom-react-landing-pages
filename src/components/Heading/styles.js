import styled, { css } from 'styled-components';

const titleSize = {
  small: (theme) => css`
    font-size: ${theme.font.sizes.small};
  `,
  medium: (theme) => css``,
  big: (theme) => css``,
};

export const Title = styled.h1`
  ${({ theme, colorDark, size }) => css`
  color: ${colorDark ? theme.colors.primaryColor : theme.colors.white};
  ${titleSize[size](theme)};
  `}
`;
