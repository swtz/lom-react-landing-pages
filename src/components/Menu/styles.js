import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    position: fixed;
    z-index: 5;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    width: 100%;
    height: 100%;
    border-bottom: ${theme.colors.mediumGray};
  `}
`;

export const MenuContainer = styled.div`
  ${({ theme }) => css`
    
  `}
`;

export const Button = styled.button`
  ${({ theme }) => css`
    
  `}
`;
