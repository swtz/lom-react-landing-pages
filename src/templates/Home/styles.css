.container {
  max-width: 1200px;
  margin: 0 auto;
  min-height: 100vh;
  background: #eee;
  padding: 30px;
}

.btn-container {
  min-height: 60px;
  display: flex;
  align-items: flex-end;
}

.search-container {
  margin-bottom: 30px;
}

.search-container h1 {
  margin-bottom: 15px;
}
