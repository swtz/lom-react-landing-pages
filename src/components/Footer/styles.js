import styled, { css } from 'styled-components';
import { Container as TextComponent } from '../TextComponent/styles';
import { Container as SectionContainer } from '../SectionContainer/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    text-align: center;
    border-top: 0.1rem solid ${theme.colors.mediumGray};

    a {
      color: inherit;
      text-decoration: none;
    }
    
    & ${TextComponent} {
      font-size: ${theme.font.sizes.small};
    }

    & ${SectionContainer} {
      padding-top: ${theme.font.sizes.small};
      padding-bottom: ${theme.font.sizes.small};
    }
  `}
`;
