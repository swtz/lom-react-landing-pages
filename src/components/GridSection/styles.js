import styled, { css } from 'styled-components';
import { Title as HeadingContainer } from '../Heading/styles';

export const Container = styled.div`
  ${({ theme }) => css``}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
  counter-reset: grid-counter;
  `}
`;

export const GridElement = styled.div`
  ${({ theme }) => css`
  ${HeadingContainer} {
    position: relative;
    left: 5rem;
  }

  ${HeadingContainer}::before {
    counter-increment: grid-counter;
    content: counter(grid-counter);
    position: absolute;
    font-size: 7rem;
    top: -3rem;
    left: -5rem;
    transform: rotate(15deg);
  }
  `}
`;
