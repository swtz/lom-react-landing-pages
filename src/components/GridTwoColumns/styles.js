import styled, { css } from 'styled-components';
import { Title as Heading } from '../Heading/styles';

export const Container = styled.div`
  ${({ theme, background }) => css`
    display: grid;
    grid-template-columns: 1fr 2fr;
    align-items: center; // alinha 'os filhos' no eixo Y
    gap: ${theme.spacings.large};

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
      text-align: center;
    }

    & ${Heading} {
      margin-bottom: ${theme.spacings.xlarge};
      color: ${background ? theme.colors.white : theme.colors.primaryColor};
    }

  `};
`;

export const TextContainer = styled.div`
  ${({ theme }) => css`
  @media ${theme.media.lteMedium} {
      margin-bottom: ${theme.spacings.large};
    }
  `};
`;

export const ImageContainer = styled.div`
  ${({ theme }) => css`
  `};
`;

export const Image = styled.img`
  ${({ theme }) => css`
    max-width: 100%;
  `};
`;
