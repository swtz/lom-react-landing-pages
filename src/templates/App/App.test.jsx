import { screen } from '@testing-library/react';
import { Home } from '.';
import { renderTheme } from '../../styles/render-theme';

test('renders <Home /> component', () => {
  const { debug } = renderTheme(<Home />);
  const headingContainer = screen.getByRole('heading', {
    name: 'Hello',
  }).parentElement;
  expect(headingContainer).toMatchSnapshot();
  expect(headingContainer).toHaveStyleRule('background', 'blue');
});
