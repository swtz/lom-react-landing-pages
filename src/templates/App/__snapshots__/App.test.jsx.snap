// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <Home /> component 1`] = `
.c0 {
  background: blue;
}

<div
  class="c0"
>
  <h1>
    Hello
  </h1>
</div>
`;
